src/protocol/ipa/mod.rs

```diff
@@ -283,13 +283,19 @@ where
 /// # Panics
 /// Propagates errors from multiplications
 #[allow(dead_code, clippy::too_many_lines)]
-pub async fn ipa_wip_malicious<F, MK, BK>(
-    sh_ctx: SemiHonestContext<'_, F>,
+pub async fn ipa_wip_malicious<'a, F, MK, BK>(
+    sh_ctx: SemiHonestContext<'a, F>,
     input_rows: &[IPAInputRow<F, MK, BK>],
     per_user_credit_cap: u32,
     max_breakdown_key: u128,
     num_multi_bits: u32,
-) -> Result<Vec<MCAggregateCreditOutputRow<F, MaliciousReplicated<F>>>, Error>
+) -> Result<
+    (
+        MaliciousValidator<'a, F>,
+        Vec<MCAggregateCreditOutputRow<F, MaliciousReplicated<F>>>,
+    ),
+    Error,
+>
```

Comment:
I don't understand why we had to change the return value of this function to bring back the test.

Why can't we call validate within `ip_wip_malicious`, and just return what we were before... and call `reconstruct` on that?

Reply:
Surely we can do that. I just wasn't sure which is a preferred way- returning maliciousreplicated and let the callers validate it (maybe in clear if they want to) or validate and return the output

Reply:
@benjaminsavage thoughts on this?